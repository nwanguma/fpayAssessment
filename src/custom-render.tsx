import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

const Wrapper: React.FC = ({ children }) => {
  return <MemoryRouter>{children}</MemoryRouter>;
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: Wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
