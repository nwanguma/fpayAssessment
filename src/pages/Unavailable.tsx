import { Link } from "react-router-dom";

import ErrorTemplate from "../templates/ErrorTemplate";

import notFound from "../assets/images/503.png";
import refresh from "../assets/images/ArrowsCounterClockwise.svg";
import clock from "../assets/images/Clock.svg";
import email from "../assets/images/EnvelopeSimple.svg";

const Unavailable: React.FC = () => {
  return (
    <ErrorTemplate image={notFound}>
      <div className="container" data-testid="container">
        <h2 className="text-[72px] leading-[90px] text-grey-large font-black">
          503
        </h2>
        <h3 className="mt-1 md:mt-3 text-[24px] leading-[30px] text-blue-black font-bold">
          Well, this is embarassing...
        </h3>
        <p className="mt-2 md:mt-3 text-[16px] leading-[24px] text-grey-text">
          Sorry this is not working properly. We know about this mistake and
          we’re working to fix it.
        </p>
        <p className="mt-2 md:mt-3 text-[16px] leading-[24px] text-grey-text">
          In the meantime, here’s what you can do:
        </p>
        <p className="mt-1 md:mt-3 flex px-3 items-center">
          <img
            className="w-[15px] h-[15px] inline-block mr-2"
            src={refresh}
            alt=""
          />
          <span className="text-[16px] leading-[24px] text-grey-text">
            Refresh the page (sometimes this helps).
          </span>
        </p>
        <p className="mt-1 md:mt-3 flex px-3 items-center">
          <img
            className="w-[15px] h-[15px] inline-block mr-2"
            src={clock}
            alt=""
          />
          <span className="text-[16px] leading-[24px] text-grey-text">
            Try again in 30 minutes.
          </span>
        </p>
        <p className="mt-1 md:mt-3 flex px-3 items-center">
          <img
            className="w-[15px] h-[15px] inline-block mr-2"
            src={email}
            alt=""
          />
          <span className="text-[16px] leading-[24px] text-grey-text">
            Email us at{" "}
            <span className="text-blue-tab">support@fliqpay.com</span> if this
            persists.
          </span>
        </p>
        <Link
          to="/"
          className="flex items-center mx-auto md:-mx-0 justify-center leading-[40px] h-[40px] w-[172px] bg-blue-tab text-white-main btn mt-8"
        >
          Refresh Page
        </Link>
      </div>
    </ErrorTemplate>
  );
};

export default Unavailable;
