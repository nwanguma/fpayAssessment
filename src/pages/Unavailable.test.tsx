import { render, screen } from "../custom-render";

import Unavailable from "./Unavailable";

describe("Unavailable", () => {
  test("<Unavailable /> matches snapshot", () => {
    const component = render(<Unavailable />);

    expect(component.container).toMatchSnapshot();
  });

  test("It renders children correctly", () => {
    const App = render(<Unavailable />);

    expect(screen.getByTestId("container")).toBeInTheDocument();
  });

  test("It renders the correct text", () => {
    render(<Unavailable />);

    expect(screen.getByText("503")).toBeInTheDocument();
  });

  test("It renders the correct link", () => {
    render(<Unavailable />);

    expect(screen.getByText("Refresh Page").closest("a")).toHaveAttribute(
      "href",
      "/"
    );
  });
});

export {};
