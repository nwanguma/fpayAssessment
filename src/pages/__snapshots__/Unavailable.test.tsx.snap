// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unavailable <Unavailable /> matches snapshot 1`] = `
<div>
  <div
    class="wrapper"
    data-testid="wrapper"
  >
    <header
      class="header--payout w-[100%] md:w-[90%] mx-auto flex items-center py-6 justify-between"
    >
      <a
        href="/"
      >
        <img
          alt="fliqpay"
          src="logo.svg"
        />
      </a>
    </header>
    <main
      class="container min-h-[80vh] min-w-full flex items-center"
    >
      <div
        class="container flex flex-col-reverse  md:flex-row justify-center md:justify-between items-center max-w-[400px] md:text-left md:max-w-[80%] mx-auto"
      >
        <div
          class="container mt-3 flex-item w-[90%] text-center md:text-left"
        >
          <div
            class="container"
            data-testid="container"
          >
            <h2
              class="text-[72px] leading-[90px] text-grey-large font-black"
            >
              503
            </h2>
            <h3
              class="mt-1 md:mt-3 text-[24px] leading-[30px] text-blue-black font-bold"
            >
              Well, this is embarassing...
            </h3>
            <p
              class="mt-2 md:mt-3 text-[16px] leading-[24px] text-grey-text"
            >
              Sorry this is not working properly. We know about this mistake and we’re working to fix it.
            </p>
            <p
              class="mt-2 md:mt-3 text-[16px] leading-[24px] text-grey-text"
            >
              In the meantime, here’s what you can do:
            </p>
            <p
              class="mt-1 md:mt-3 flex px-3 items-center"
            >
              <img
                alt=""
                class="w-[15px] h-[15px] inline-block mr-2"
                src="ArrowsCounterClockwise.svg"
              />
              <span
                class="text-[16px] leading-[24px] text-grey-text"
              >
                Refresh the page (sometimes this helps).
              </span>
            </p>
            <p
              class="mt-1 md:mt-3 flex px-3 items-center"
            >
              <img
                alt=""
                class="w-[15px] h-[15px] inline-block mr-2"
                src="Clock.svg"
              />
              <span
                class="text-[16px] leading-[24px] text-grey-text"
              >
                Try again in 30 minutes.
              </span>
            </p>
            <p
              class="mt-1 md:mt-3 flex px-3 items-center"
            >
              <img
                alt=""
                class="w-[15px] h-[15px] inline-block mr-2"
                src="EnvelopeSimple.svg"
              />
              <span
                class="text-[16px] leading-[24px] text-grey-text"
              >
                Email us at
                 
                <span
                  class="text-blue-tab"
                >
                  support@fliqpay.com
                </span>
                 if this persists.
              </span>
            </p>
            <a
              class="flex items-center mx-auto md:-mx-0 justify-center leading-[40px] h-[40px] w-[172px] bg-blue-tab text-white-main btn mt-8"
              href="/"
            >
              Refresh Page
            </a>
          </div>
        </div>
        <div
          class="container mx-auto md:mx-0 w-[250px] h-[150px] md:w-[350px] md:h-[250px]"
        >
          <img
            alt="error pic"
            class="w-[100%] h-[100%]"
            src="503.png"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
